src/commcare_cloud/fab/utils.py

```diff
@@ -140,11 +141,7 @@ def deploy_ref(self):
             return env.code_branch
 
         # turn whatever `code_branch` is into a commit hash
-        try:
-            branch = self.repo.get_branch(self._code_branch)
-            return branch.commit.sha
-        except GithubException:
-            return self.repo.get_commit(sha=self._code_branch).sha
+        return sh.git.bake(_tty_out=False)("rev-parse", self._code_branch).strip()
```

Comment:
I am sorry for leading you astray. I don't think this will work since this will most likely operate on the local repository (`commcare-cloud`), not `dimagi/commcare-hq` referenced by `self.repo`. I will see if I can come up with a better option...

Reply:
That is a good point, this isn't acting in the right repo. Curiously, this does run without error, although I only tested enough to see it didn't throw an error, I didn't do a full deploy.

pygithub allows you to list a repo's tags; that's the approach I was considering when I said I didn't want to bother adding support for tags.

Reply:
I think this will work (tested locally with branch name, commit hash, and tag name)
```suggestion
        return self.repo.get_commit(self._code_branch).sha
```